csrc/keyutils.cpp

```diff
@@ -170,35 +168,36 @@ const EVP_MD* digestFromJstring(raii_env& env, jstring digestName)
     return result;
 }
 
-RSA* RSA_new_private_key_no_e(BIGNUM* n, BIGNUM* d)
+RSA* new_private_RSA_key_with_no_e(BIGNUM const* n, BIGNUM const* d)
 {
+#ifdef FIPS_BUILD
+    // AWS-LC-FIPS doesn't have RSA_new_private_key_no_e method yet.
+    // The following implementation has been copied from AWS-LC:
+    // https://github.com/aws/aws-lc/blob/v1.30.1/crypto/fipsmodule/rsa/rsa.c#L147
     RSA_auto rsa = RSA_auto::from(RSA_new());
     if (rsa.get() == nullptr) {
         throw_openssl("RSA_new failed");
     }
-    // The FIPS builds of AWS-LC do not have an API to disable blinding.
-#ifdef FIPS_BUILD
 
     // RSA struct is not opaque in FIPS mode.
     rsa->flags |= RSA_FLAG_NO_BLINDING;
 
+    bn_dup_into(&rsa->n, n);
+    bn_dup_into(&rsa->d, d);
+
+    return rsa.take();
+
 #else
 
-    // RSA_blinding_off_temp_for_accp_compatibility is marked deprecated, so we need to disable the
-    // "deprecated-declarations" check for this invocation.
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-    // We need to change this API once AWS-LC provides a method similar to the following:
-    // https://github.com/google/boringssl/blob/master/include/openssl/rsa.h#L630
-    RSA_blinding_off_temp_for_accp_compatibility(rsa);
-#pragma GCC diagnostic pop
+    RSA* result = RSA_new_private_key_no_e(n, d);
```

Comment:
why is using this method preferable to doing it the FIPS way with the `dup_into`s?

Reply:
In non-FIPS, the RSA structure is opaque and we don't have a way to turn off blinding. This method creates a private RSA key and sets the blinding flag properly. I think in the next FIPS release, the RSA structure will become opaque and such a method will become available.